common/components/common/tags/TagSelectorCollapsible.jsx

```diff
@@ -57,21 +58,35 @@ class TagSelectorCollapsible extends React.Component<Props, State> {
   }
 
 
-  selectTag(tag: TagDefinition): void {
-    var tagInState = _.has(this.state.selectedTags, tag.tag_name);
-    //if tag is NOT currently in state, add it, otherwise remove
-    if(!tagInState) {
-      ProjectSearchDispatcher.dispatch({
-        type: 'ADD_TAG',
-        tag: tag.tag_name,
+  selectTag(tag: TagDefinition, opts): void {
+    //opts is optional, if passed in expect opts to be an object formatted like { multiple: true, type: ACTION_TO_TAKE }
+    var opts = opts || {}; //to avoid undefined error, create an empty object if opts isn't passed in. TODO: make this work with let instead of var
+    if (opts.multiple && opts.type) {
+      let tagnames = tag.map(function(t) {
+          return t.tag_name
       });
-      metrics.logSearchFilterByTagEvent(tag);
-    } else {
       ProjectSearchDispatcher.dispatch({
-        type: 'REMOVE_TAG',
-        tag: tag,
+        type: opts.type,
+        tag: tagnames,
       });
+    } else {
+      metrics.logSearchFilterByTagEvent(tag);
```

Comment:
I think I've got this in the right place but we'll need to review metrics for the add-many-tags event in any case